test/unittests/test_extend_tt.cpp

```diff
@@ -0,0 +1,121 @@
+#include "algorithms/synthesis/encoding.hpp"
+#include "core/io/pla_parser.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TruthTableExtend: public testing::Test {
+protected:
+    TruthTable  tt{};
+    std::string test_circuits_dir = "./circuits/";
+
+    TruthTable::Cube c00{};
+    TruthTable::Cube c11{};
+
+    TruthTable::Cube c000{};
+    TruthTable::Cube c001{};
+    TruthTable::Cube c010{};
+    TruthTable::Cube c011{};
+    TruthTable::Cube c100{};
+    TruthTable::Cube c101{};
+    TruthTable::Cube c110{};
+    TruthTable::Cube c111{};
+
+    void SetUp() override {
+        c00 = TruthTable::Cube::fromInteger(0b00U, 2U);
+        c11 = TruthTable::Cube::fromInteger(0b11U, 2U);
+
+        c000 = TruthTable::Cube::fromInteger(0b000U, 3U);
+        c001 = TruthTable::Cube::fromInteger(0b001U, 3U);
+        c010 = TruthTable::Cube::fromInteger(0b010U, 3U);
+        c011 = TruthTable::Cube::fromInteger(0b011U, 3U);
+        c100 = TruthTable::Cube::fromInteger(0b100U, 3U);
+        c101 = TruthTable::Cube::fromInteger(0b101U, 3U);
+        c110 = TruthTable::Cube::fromInteger(0b110U, 3U);
+        c111 = TruthTable::Cube::fromInteger(0b111U, 3U);
```

Comment:
I'd propose to introduce a
```c++
auto equals(std::uint64_t num) -> bool
```
method in the `Cube` class.
Additionally, a
```c++
auto find(std::uint64_t num) -> decltype(cubeMap.begin())
```
method in the `TruthTable` class might be helpful that creates a corresponding cube and calls `find()` on the underlying `cubeMap`.

Then all of these variables can be removed (also in the other test files) and the respective `EXPECT_EQ(cube, cXYZ)` calls could just be replaced with `EXPECT_TRUE(cube.equals(0bXYZU))`.

Reply:
done 👍🏼